homeassistant/components/growatt_server/sensor.py

```diff
@@ -46,8 +47,7 @@ def get_device_list(api, config):
         not login_response["success"]
         and login_response["msg"] == LOGIN_INVALID_AUTH_CODE
     ):
-        _LOGGER.error("Username, Password or URL may be incorrect!")
-        return
+        raise HomeAssistantError("Username, Password or URL may be incorrect!")
```

Comment:
```suggestion
        raise ConfigEntryError("Username, Password or URL may be incorrect!")
```

I think this is the error that should be raised

Reply:
I know for sure if it hits this one that it shows somewhere in the UI